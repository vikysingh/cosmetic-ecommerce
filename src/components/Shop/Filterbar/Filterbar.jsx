import React, { useRef } from "react"

import { FilterOutlined, CloseOutlined } from "@ant-design/icons"

import styles from "./Filterbar.module.css"
import globalStyles from "../../../styles/Globals.module.css"

import Filter from "../Filter/Filter"
import PriceFilter from "../PriceFilter/PriceFilter"

import ProductStatics from "../../../constants/ProductStatics.json"


export default function Filterbar() {
    
    const filterBarRef = useRef(null)

    function openFilters() {
        filterBarRef.current.style.display = "block"
    }

    function closeFilters() {
        filterBarRef.current.style.display = "none"
    }


    return <div className={styles.FILTER_BAR}  >

        <div id={styles.FILTER_BAR__head} className={globalStyles.flexRowEndCenter}>
            <label> Refine search </label>
            <FilterOutlined id={styles.FILTER_BAR__open_icon}
            onClick={openFilters}/>
        </div>

        <div className={styles.FILTER_BAR__body} ref={filterBarRef}>
            <CloseOutlined id={styles.FILTER_BAR__close_icon}
            onClick={closeFilters} />
            <Filter title="Products" filterList={ProductStatics.productType} type="products"  />
            <Filter title="Brands" filterList={ProductStatics.brands} type="brand"   />
            <PriceFilter />
        </div>

    </div>
}