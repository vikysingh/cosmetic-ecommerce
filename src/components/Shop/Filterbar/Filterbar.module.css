.FILTER_BAR {
    width: 100%;
    height: 100%;
    border-right: 1px solid #000;
    background-color: #eee;
}

#FILTER_BAR__head {
    width: 100%;
    height: 40px;
    display: none;
}

#FILTER_BAR__head label {
    opacity: 0.8;   
}

#FILTER_BAR__open_icon {
    font-size: 1.3rem;
    float: right;
    margin-right: 15px;
    padding: 10px;
    height: 100%;
    /* display:none; */
}

#FILTER_BAR__close_icon {
    display: none;
}

.FILTER_BAR__open_icon_active {
    background-color: #fff;
    border-radius: 50px;
}

@media (max-width: 600px) {

    .FILTER_BAR__body {
        position: fixed;
        width: 100vw;
        height: 100vh;
        top: 0;
        z-index: 5;
        background-color: #eee;
        display: none;
    }

    #FILTER_BAR__head {
        display: flex;
    }

    #FILTER_BAR__close_icon {
        display: block;
        float: right;
        margin-top: 15px;
        margin-right: 15px;
        font-size: 1rem;
        padding: 10px;
        background-color: #fff;
        border-radius: 50px;
    }
}