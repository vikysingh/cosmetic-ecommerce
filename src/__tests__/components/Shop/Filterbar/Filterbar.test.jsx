import React from "react"
import { shallow } from "enzyme"

import Filterbar from "../../../../components/Shop/Filterbar/Filterbar"

test('Should render Filterbar from Shop', () => {
    const wrapper = shallow(<Filterbar />)

    expect(wrapper).toMatchSnapshot()
})