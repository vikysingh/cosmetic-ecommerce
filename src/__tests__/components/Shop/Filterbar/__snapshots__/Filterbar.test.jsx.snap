// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Filterbar from Shop 1`] = `
<div
  className="FILTER_BAR"
>
  <div
    className="flexRowEndCenter"
    id="FILTER_BAR__head"
  >
    <label>
       Refine search 
    </label>
    <[object Object]
      id="FILTER_BAR__open_icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="FILTER_BAR__body"
  >
    <[object Object]
      id="FILTER_BAR__close_icon"
      onClick={[Function]}
    />
    <[object Object]
      filterList={
        Array [
          "Blush",
          "Bronzer",
          "Eyebrow",
          "Eyeliner",
          "Eyeshadow",
          "Foundation",
          "Lip Liner",
          "Lipstick",
          "Mascara",
          "Nail polish",
        ]
      }
      title="Products"
      type="products"
    />
    <[object Object]
      filterList={
        Array [
          "Benefit",
          "Covergirl",
          "Dalish",
          "Essie",
          "Fenty",
          "Glossier",
          "Iman",
          "L'oreal",
          "Maybelline",
          "Orly",
        ]
      }
      title="Brands"
      type="brand"
    />
    <[object Object] />
  </div>
</div>
`;
